src/DataStructures/Tensor/Expressions/TensorExpression.hpp

```diff
@@ -435,6 +439,131 @@ struct TensorExpression<Derived, DataType, Symm, IndexList, ArgsList<Args...>>
+  /// TensorIndex::values that correspond to generic tensor indices,
+  /// `lhs_tensor_multi_index` is a multi-index for the LHS tensor, and the
+  /// equivalent RHS tensor multi-index is returned. If we have LHS tensor
+  /// \f$L_{ab}\f$, RHS tensor \f$R_{ba}\f$, and the LHS component \f$L_{31}\f$,
+  /// the corresponding RHS component is \f$R_{13}\f$.
+  ///
+  /// Here is an example of what the algorithm does:
+  ///
+  /// `lhs_index_order`:
+  /// \code
+  /// [0, 1, 2] // i.e. abc
+  /// \endcode
+  /// `rhs_index_order`:
+  /// \code
+  /// [1, 2, 0] // i.e. bca
+  /// \endcode
+  /// `lhs_tensor_multi_index`:
+  /// \code
+  /// [4, 0, 3] // i.e. a = 4, b = 0, c = 3
+  /// \endcode
+  /// returned RHS tensor multi-index:
+  /// \code
+  /// [0, 3, 4] // i.e. b = 0, c = 3, a = 4
+  /// \endcode
+  ///
+  /// \tparam NumberOfIndices the number of indices in the RHS tensor
+  /// \param lhs_index_order the generic index order of the LHS tensor
+  /// \param rhs_index_order the generic index order of the RHS tensor
+  /// \param lhs_tensor_multi_index the specific LHS tensor multi-index
+  /// \return the RHS tensor multi-index that corresponds to
+  /// `lhs_tensor_multi_index`, according to the index orders in
+  /// `lhs_index_order` and `rhs_index_order`
+  template <size_t NumberOfIndices>
+  SPECTRE_ALWAYS_INLINE static constexpr std::array<size_t, NumberOfIndices>
+  compute_rhs_tensor_index(
+      const std::array<size_t, NumberOfIndices>& lhs_index_order,
+      const std::array<size_t, NumberOfIndices>& rhs_index_order,
+      const std::array<size_t, NumberOfIndices>&
+          lhs_tensor_multi_index) noexcept {
+    std::array<size_t, NumberOfIndices> rhs_tensor_multi_index{};
+    for (size_t i = 0; i < NumberOfIndices; ++i) {
+      rhs_tensor_multi_index[static_cast<unsigned long>(
+          std::distance(rhs_index_order.begin(),
+                        alg::find(rhs_index_order, lhs_index_order[i])))] =
+          lhs_tensor_multi_index[i];
+    }
+    return rhs_tensor_multi_index;
+  }
+
+  /// \brief Computes a mapping from the storage indices of the left hand side
+  /// tensor to the right hand side tensor
+  /// \tparam LhsStructure the Structure of the Tensor on the left hand side of
+  /// the TensorExpression
+  /// \tparam LhsIndices the TensorIndexs of the Tensor on the left hand side
+  /// \return the mapping from the left hand side to the right hand side storage
+  /// indices
+  template <typename LhsStructure, typename... LhsIndices>
+  SPECTRE_ALWAYS_INLINE static constexpr std::array<size_t,
+                                                    LhsStructure::size()>
+  compute_lhs_to_rhs_map() noexcept {
+    constexpr size_t num_components = LhsStructure::size();
+    constexpr size_t rank = LhsStructure::rank();
+    std::array<size_t, num_components> lhs_to_rhs_map{};
+    auto LhsStorageToTensorIndices = LhsStructure::storage_to_tensor_index();
```

Comment:
Not sure why this is camel case. I will change this to `lhs_storage_to_tensor_indices` as part of a fixup

Reply:
Done!